src/fontcache.cpp

```diff
@@ -147,6 +147,30 @@ static void LoadFontHelper(FontSize fs, const std::string &file_name, uint size)
 #endif
 }
 
+void ResizeFont(FontSize font_size, uint size)
+{
+	FontCacheSubSetting *setting = GetFontCacheSubSetting(font_size);
+
+	if (setting->size == size) {
+		return;
+	}
+
+	setting->size = size;
+
+	// Default fonts are empty here. We will allow the user to resize the default font:
+	if (setting->font.empty()){
+		TryLoadDefaultTrueTypeFont(font_size, size);
+	} else {
+		LoadFontHelper(font_size);
+	}
+
+	LoadStringWidthTable();
+	UpdateAllVirtCoords();
+	ReInitAllWindows(true);
+
+	if (_save_config) SaveToConfig();
+}
```

Comment:
I think this will replace the FontCache for font_size without destroying the existing FontCache.

Resizing a font shouldn't need to load fonts again, probably you can expose SetFontSize() which all scaleable FontCaches implement.

Reply:
Done.